### DISCUSSION TITLE
Hostname mapping failure for introspection

### DISCUSSION THREAD

**Discussion by EML-github** [keycloak/keycloak#27004] [Q&A]
I have a working setup, and I'd now like to use the introspection endpoint. The current working setup is:

1. Keycloak proxied (`proxy=edge`) behind Apache, on `http://localhost:8095`
2. `hostname-url=https://example.com/keycloak`
3. `hostname-admin-url=https://example.com/keycloak`

The Apache proxy settings are:
```
  <Location /keycloak/>
    ProxyPass        http://localhost:8095/
    ProxyPassReverse http://localhost:8095/
    Header           unset X-Frame-Options
  </Location>
```
So the proxy converts accesses to `https://example.com/keycloak` to `http://localhost:8095`, but the `hostname-url` setting makes Keycloak think it's still at `https://example.com/keycloak`.

The `hostname-debug` page confirms this:

```
Frontend                  https://example.com/keycloak [OK]
Backend 	          https://example.com/keycloak [OK]
Admin 	                  https://example.com/keycloak [OK]
hostname-url 	          https://example.com/keycloak
hostname-admin-url        https://example.com/keycloak
hostname-strict 	  true
hostname-strict-backchannel false
```
So far so good, but there's an introspection problem. If a user requests a token, the token sets the issuer to `https://example.com/keycloak/realms/etc`, as expected. If the user later accesses the introspection endpoint to check this token, they use this URL:

`https://example.com/keycloak/realms/vserver/protocol/openid-connect/token/introspect`

However, the proxy setting turns this into `/realms/vserver/protocol/openid-connect/token/introspect`, *without* the `/keycloak`. Authorisation now fails because this differs from the issuer URL.

In short: Keycloak knows that backend accesses are effectively to `https://example.com/keycloak`,  but this knowledge doesn't seem to reach the introspection endpoint.

Does this make sense? Any suggestions for a fix?

**Comment by EML-github** [comment]
False alarm - introspection failed because of other errors in the comms from Apache to Keycloak. The difference in the URLs turned out not to be a problem.

**Reply by codespearhead** [reply]
@EML-github Can you [close this discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion) then?
